src/services/userService.ts

```diff
@@ -13,17 +11,29 @@ const users: Array<gqlTypes.User> = []
 export async function getUserById(id: string)
     : Promise<Result<gqlTypes.User>> {
     try {
-        const selectedUser = users.find(u => u.id === id)
+        const { data } = await supabase
+            .from('user')
+            .select('*')
+            .eq('id', id) 
 
-        if (!selectedUser) {
+        if (!data) {
             throw new Error(`No user found with id: ${id}`)
         }
+
+        const selectedUser:gqlTypes.User = {
```

Comment:
you can pull `data[0]` out into a separate variable to describe what that array value is and will save repeating the array accessor